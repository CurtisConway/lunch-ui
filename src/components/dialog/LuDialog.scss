@import '../../assets/sass/_variables';
@import '../../assets/sass/_mixins';

.lu-dialog {
  display: none;
  position:fixed;
  flex-direction:column;
  align-items:center;
  justify-content:center;
  top:0;
  left:0;
  bottom:0;
  right:0;
  z-index:100;

   &.dialog-active {
     display: flex;
   }

  .dialog-foreground {
    position:relative;
    width:95%;
    max-width:95%;
    min-height:300px;
    z-index:2;

    @include small {
      max-width: 640px;
    }

    @include medium {
      max-width: 960px;
    }

    @include large {
      max-width:1200px;
    }

    @include xLarge {
      max-width:1640px;
    }
  }

  .dialog-background {
    position:absolute;
    cursor:pointer;
    background-color:rgba(0,0,0,.3);
    top:0;
    left:0;
    bottom:0;
    right:0;
    z-index:1;
  }
}