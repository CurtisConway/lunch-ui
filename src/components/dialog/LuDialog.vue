<template>
  <div
    :class="defaultClassList"
    role="dialog"
    aria-modal="true"
  >
    <LuRow class="dialog-foreground">
      <transition name="slide-up-fade">
        <LuCard
          v-show="value"
          bgColor="white"
          borderRadius="2"
        >
          <slot></slot>
        </LuCard>
      </transition>
    </LuRow>

    <span
      class="dialog-background"
      @click="closeDialog">
    </span>
  </div>
</template>

<script>
import './LuDialog.scss';
import LuRow from '../grid/LuRow.vue';
import LuCard from '../card/LuCard.vue';

export default {
  name: 'LuDialog',
  components: {
    LuRow,
    LuCard,
  },
  props: {
    value: {
      type: Boolean,
      default: () => false,
    },
  },
  watch: {
    value(value) {
      if (value) {
        this.$emit('dialogOpen', this);
        if (document.addEventListener) {
          document.addEventListener('keyup', this.closeDialogWithEscape);
        }
      } else {
        this.$emit('dialogClose', this);
        if (document.removeEventListener) {
          document.removeEventListener('keyup', this.closeDialogWithEscape);
        }
      }
    },
  },
  computed: {
    defaultClassList() {
      return [
        'lu-dialog',
        {
          'dialog-active': this.value,
        },
      ];
    },
  },
  methods: {
    closeDialog() {
      this.$emit('input', false);
    },
    closeDialogWithEscape(event) {
      if (event.key === 'Escape') {
        this.closeDialog();
      }
    },
  },
};
</script>
