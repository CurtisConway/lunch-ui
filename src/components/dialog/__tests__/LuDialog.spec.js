import Vue from 'vue';
import Router from 'vue-router';

import LuDialog from '../LuDialog.vue';

import {
  createLocalVue,
  mount,
} from '@vue/test-utils';
import { compileToFunctions } from 'vue-template-compiler'

describe('LuDialog.vue', () => {
  let mountFunction;
  let router = Router;
  let localVue = typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);

    mountFunction = (options = {}) => {
      return mount(LuDialog, {
        localVue,
        router,
        ...options,
      });
    }
  })

  it('should render component and match snapshot', () => {
    expect(mountFunction().html()).toMatchSnapshot();
  });

  it('should emit an openDialog event', async () => {
    const wrapper = mountFunction({
      value: false,
    })
    const event = jest.fn();
    wrapper.vm.$on('dialogOpen', event);

    wrapper.setProps({
      value: true,
    });

    await wrapper.vm.$nextTick()
    expect(event.mock.calls).toHaveLength(1);
  });

  it('should emit a closeDialog event', async () => {
    const wrapper = mountFunction({
      value: false,
    })
    const event = jest.fn();
    wrapper.vm.$on('dialogClose', event);

    wrapper.setProps({
      value: true,
    });
    await wrapper.vm.$nextTick()
    wrapper.setProps({
      value: false,
    });
    await wrapper.vm.$nextTick()
    expect(event.mock.calls).toHaveLength(1);
  });
});
