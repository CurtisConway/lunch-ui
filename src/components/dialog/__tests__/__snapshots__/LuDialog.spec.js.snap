// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LuDialog.vue should render component and match snapshot 1`] = `
<div role="dialog" aria-modal="true" class="lu-dialog">
  <div class="dialog-foreground flex flex-row flex-wrap">
    <div class="flex flex-column lu-card elevation-4 rad-2 bg-white" style="display: none;"></div>
  </div> <span class="dialog-background"></span>
</div>
`;
